File: Python/1009. Complement of base 10 integer.py
class Solution(object):
    def bitwiseComplement(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 1

        m = n
        mask = 0

        while m != 0:
            mask = (mask << 1) | 1
            m = m >> 1

        return (~n) & mask

File: JavaScript/1009. Complement of base 10 integer.js
/**
 * @param {number} n
 * @return {number}
 */
var bitwiseComplement = function (n) {
  if (n === 0) {
    return 1;
  }
  let m = n;
  let mask = 0;

  while (m != 0) {
    mask = (mask << 1) | 1;
    m = m >> 1;
  }

  let answer = ~n & mask;
  return answer;
};
